up-core-api/uprotocol/uri.proto

```diff
@@ -54,46 +54,30 @@ message UUri {
 // An Authority represents the deployment location of a specific Software Entity.
 // Authority can be represented in either a name (i.e example.com), ip address (205.236.147.1)
 // or an ID (i.e. VIN, SHA 128, or any other identifier that is less than 255 bytes.
-// *NOTE:* When Authority is empty (neither name, ip, or id are set) the authority
-// is local.
 message UAuthority {
-  optional string name = 1; // Domain & device name as a string
-  oneof number {
-    bytes ip = 2; 	// IPv4 or IPv6 Address in byte format
-    bytes id = 3;		// Unique ID for the device, could be a VIN, SHA 128, or any other identifier
-                    // *NOTE:* MAX length is 255 bytes
-  }
+  string name = 1; // Domain & device name as a string
 }
 
 
 // Data representation of an <b> Software Entity - uE</b><br>
-// Software entities are distinguished by using a unique name and id
+// Software entities are distinguished by using a unique id 
 // along with the specific version of the software.
 // An Software Entity is a piece of software deployed somewhere on a uDevice.<br>
 // The Software Entity is used in the source and sink parts of communicating software.<br>
 // A uE that publishes events is a <b>Service</b> role.<br>
 // A uE that consumes events is an <b>Application</b> role.
 message UEntity {
-  optional string name = 1;          // Name of the entity
-  optional uint32 id = 2;            // The numeric ID for the uEntity assigned from the name/number registry
-  optional uint32 version_major = 3; // optional major version of the uEntity
-  optional uint32 version_minor = 4; // optional minor version of the uEntity
+  optional uint32 id = 1;            // UEntity ID
+  optional uint32 version_major = 2; // optional major version of the uEntity
+  optional uint32 version_minor = 3; // optional minor version of the uEntity
```

Comment:
I haven't seen any place yet where minor version is being used. Is this really needed?

Reply:
I have removed it. Minor is really only needed at the application layer.